src/features/defi/providers/idle/components/IdleManager/Overview/IdleEmpty.tsx

```diff
@@ -0,0 +1,226 @@
+      bullets: [
+        'idle.emptyBody.leverageYield',
+        'idle.emptyBody.higherRisk',
+        'idle.emptyBody.noLockingPeriod',
+        'idle.emptyBody.autoReinvest',
+      ],
+    },
+  ],
+  [IdleTag.SeniorTranche]: [
+    {
+      title: 'idle.emptyBody.seniorTranche.page-1.title',
+      description: 'idle.emptyBody.seniorTranche.page-1.body',
+      icon: <Image src={SeniorTrancheIcon} />,
+    },
+    {
+      title: 'idle.emptyBody.seniorTranche.page-2.title',
+      bullets: [
+        'idle.emptyBody.minRisk',
+        'idle.emptyBody.builtInCoverage',
+        'idle.emptyBody.noLockingPeriod',
+        'idle.emptyBody.autoReinvest',
+      ],
+    },
+  ],
+}
+
+type IdleEmptyProps = {
+  assetId: AssetId
+  onClick?: () => void
+  tags?: IdleTag[]
+  apy: string
+}
+
+export const IdleEmpty = ({ assetId, onClick, tags, apy }: IdleEmptyProps) => {
+  const translate = useTranslate()
+  const { fiatRamps } = useModal()
+  const { open: openFiatRamp } = fiatRamps
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const filter = useMemo(() => ({ assetId }), [assetId])
+  const bgImage = useColorModeValue('none', IdleBg)
+  const assetSupportsFiatRamp = useAppSelector(s => selectSupportsFiatRampByAssetId(s, filter))
+  const cryptoBalance =
+    useAppSelector(state => selectPortfolioCryptoHumanBalanceByFilter(state, filter)) ?? '0'
+  const hasZeroCryptoBalance = bnOrZero(cryptoBalance).eq(0)
+  const textShadow = useColorModeValue(
+    '--chakra-colors-blackAlpha-50',
+    '--chakra-colors-blackAlpha-400',
+  )
+
+  const handleAssetBuyClick = useCallback(() => {
+    openFiatRamp({ assetId, fiatRampAction: FiatRampAction.Buy })
+  }, [assetId, openFiatRamp])
+
+  const renderFooter = useMemo(() => {
+    return (
+      <Flex flexDir='column' gap={4} width='full'>
+        {hasZeroCryptoBalance && assetSupportsFiatRamp && (
+          <Alert status='info' justifyContent='space-between' borderRadius='xl'>
+            <Flex gap={2} alignItems='center'>
+              <AssetIcon assetId={asset?.assetId} size='sm' />
+              <Text
+                fontWeight='bold'
+                letterSpacing='-0.02em'
+                translation={['common.needAsset', { asset: asset?.name }]}
+              />
+            </Flex>
+            <Button variant='ghost' size='sm' colorScheme='blue' onClick={handleAssetBuyClick}>
+              {translate('common.buyNow')}
+            </Button>
+          </Alert>
+        )}
+        <Button size='lg' width='full' colorScheme='blue' onClick={onClick}>
+          <Text translation='common.continue' />
+        </Button>
+      </Flex>
+    )
+  }, [
+    asset?.assetId,
+    asset?.name,
+    assetSupportsFiatRamp,
+    hasZeroCryptoBalance,
+    handleAssetBuyClick,
+    onClick,
+    translate,
+  ])
+
+  const renderTags = useMemo(() => {
+    return tags?.map(tag => {
+      if (idleTagToOnboardingSteps[tag]) {
```

Comment:
Nitpick but I'd invert the flow here, early returns vs. if/else is easier to visually grap